src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -98,19 +114,26 @@ private void checkHealth() {
     }
   }
 
-  public void stop() {
+  public synchronized void stop() {
```

Comment:
So awaitShutdown() should be safe to call multiple times and should be thread safe.  awaitShutdown() does not need to check the started state.  stop also should be safe to call multiple times.  The problem is that healthCheck(), stop() and awaitShutdown() can all get called concurrently. It's not clear to me what stop() should do if start() was not called.  Since stop() can be called in a exit handling thread it's possible it can be called before start().  Should we prevent start from running if stop() was called before start()?